Duplicati/Library/Main/Database/LocalDatabase.cs

```diff
@@ -737,42 +737,34 @@ public IDictionary<string, string> GetDbOptions(System.Data.IDbTransaction trans
             return GetDbOptionList(transaction).ToDictionary(x => x.Key, x => x.Value);
         }
 
-        public bool RepairInProgress
+        private void UpdateDbOption(string key, bool value)
         {
-            get
-            {
-                return GetDbOptions().ContainsKey("repair-in-progress");
-            }
-            set
-            {
-                var opts = GetDbOptions();
+            var opts = GetDbOptions();
 
-                if (value)
-                    opts["repair-in-progress"] = "true";
-                else
-                    opts.Remove("repair-in-progress");
+            if (value)
+                opts[key] = "true";
+            else
+                opts.Remove(key);
 
-                SetDbOptions(opts);
-            }
+            SetDbOptions(opts);
         }
 
-        public bool PartiallyRecreated
+        public bool RepairInProgress
         {
-            get
-            {
-                return GetDbOptions().ContainsKey("partially-recreated");
-            }
-            set
-            {
-                var opts = GetDbOptions();
+            get => GetDbOptions().ContainsKey("repair-in-progress");
+            set => UpdateDbOption("repair-in-progress", value);
+        }
 
-                if (value)
-                    opts["partially-recreated"] = "true";
-                else
-                    opts.Remove("partially-recreated");
+        public bool PartiallyRecreated
+        {
+            get => GetDbOptions().ContainsKey("partially-recreated");
+            set => UpdateDbOption("partially-recreated", value);
+        }
 
-                SetDbOptions(opts);
-            }
+        public bool TerminatedWithActiveUploads

```

Comment:
Same as before, shouldn't there be a `TerminatedWithActiveDownloads`?

Reply:
The flag can be renamed, but the intention is to record if the remote storage is being altered and the process terminates, as we can then expect the remote storage to be partially updated. Since downloads do not alter the remote storage state, they are not considered for this flag.

Reply:
I see. My thought was like the download failed reporting, that this could be used to indicate that something went wrong.